pkg/metrics/metrics.go

```diff
@@ -0,0 +1,138 @@
+package metrics
+
+import (
+	logging "github.com/op/go-logging"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	subsystem = "asb"
+)
+
+var (
+	sandboxCreated = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Subsystem: subsystem,
+			Name:      "sandbox_created",
+			Help:      "Counter of all sandbox namespaces that are created.",
+		})
+
+	sandboxDeleted = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Subsystem: subsystem,
+			Name:      "sandbox_deleted",
+			Help:      "Counter of all sandbox namespaces that are deleted.",
+		})
+
+	specsLoaded = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: subsystem,
+			Name:      "specs_loaded",
+			Help:      "Specs loaded from registries, partitioned by registry name.",
+		}, []string{"registry_name"})
+
+	specsReset = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Subsystem: subsystem,
+			Name:      "specs_reset",
+			Help:      "Counter of how many times the specs have been reset.",
+		})
+
+	provisionJob = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Subsystem: subsystem,
+			Name:      "provision_jobs",
+			Help:      "How many provision jobs are actively in the buffer.",
+		})
+
+	deprovisionJob = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Subsystem: subsystem,
+			Name:      "deprovision_jobs",
+			Help:      "How many deprovision jobs are actively in the buffer.",
+		})
+
+	log = logging.MustGetLogger("metrics")
+)
+
+func init() {
+	prometheus.MustRegister(sandboxCreated)
+	prometheus.MustRegister(sandboxDeleted)
+	prometheus.MustRegister(specsLoaded)
+	prometheus.MustRegister(specsReset)
+	prometheus.MustRegister(provisionJob)
+	prometheus.MustRegister(deprovisionJob)
+}
+
+// Init - Initialize the metrics package.
+func Init(logger *logging.Logger) {
+	log = logger
+}
+
+func recoverMetricPanic() {
+	if r := recover(); r != nil {
+		log.Errorf("Recovering from metric function - %v", r)
+	}
+}
+
+// SandboxCreated - Counter for how many sandbox created.
+func SandboxCreated() {
+	defer recoverMetricPanic()
+	sandboxCreated.Inc()
```

Comment:
I'm going to assume that `Inc` can handle multiple things incrementing it at the same time and that we won't have a miscount due to a race condition. 🤞

Reply:
> All exported functions and methods are safe to be used concurrently unless specified otherwise.

https://godoc.org/github.com/prometheus/client_golang/prometheus


:crossed_fingers: sounds like it should be fine.